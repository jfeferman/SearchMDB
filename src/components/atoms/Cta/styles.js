import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledButton = styled.button`
  border-radius: ${px2rem(10)};
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  background: ${colors.lightGray};
  color: ${colors.black};
  font-size: ${px2rem(15)};
  line-height: ${px2rem(17)};
  letter-spacing: ${px2rem(0.34)};
  padding: ${px2rem(16)} ${px2rem(41.5)};
  vertical-align: middle;
  outline: 0;
  transition: color .15s ease-in-out,
    background-color .15s ease-in-out,
    border-color .15s ease-in-out;

  &:focus {
    box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
  }

  ${(props) => props.disabled && `
    cursor: not-allowed;
    background: ${colors.disabled};
    color: ${colors.disabledText};
    border: 0;
    &:hover {
      color: ${colors.disabledText};
    }
  `};
`;
