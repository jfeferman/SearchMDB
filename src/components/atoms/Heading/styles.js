import styled from 'styled-components';
import { fonts } from '../../styleUtils/globalStyleVariables';

export const StyledH1 = styled.h1`
  ${fonts.H1}
`;

export const StyledH2 = styled.h2`
  ${fonts.H2}
`;

export const StyledH3 = styled.h3`
  ${fonts.H3}
`;

export const StyledH4 = styled.h4`
  ${fonts.H4}
`;
