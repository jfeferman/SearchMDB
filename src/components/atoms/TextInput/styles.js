import styled from 'styled-components';
import { px2rem, colors } from '../../styleUtils/globalStyleVariables';

export const StyledTextInput = styled.input`
  width: 100%;
  box-sizing: border-box;
  height: ${px2rem(56)};
  border: 1px solid ${colors.lightGray};
  border-radius: ${px2rem(3)};
  background-color: ${colors.white};
  padding-left: ${px2rem(20)};
  font-size: ${px2rem(16)};
`;
