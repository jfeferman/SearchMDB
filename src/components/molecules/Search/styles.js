import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearch = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  width: ${px2rem(600)};

  div:nth-of-type(1) {
    margin-right: ${px2rem(10)};
  }
`;
