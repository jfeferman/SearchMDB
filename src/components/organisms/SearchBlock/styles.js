import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledSearchBlock = styled.div`
  margin: ${px2rem(100)} auto ${px2rem(100)} auto;
  
  h1 {
    text-align: center;
  }
`;
