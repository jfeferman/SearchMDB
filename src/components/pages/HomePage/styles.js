import styled from 'styled-components';
import { px2rem } from '../../styleUtils/globalStyleVariables';

export const StyledResultsBlock = styled.div`
  margin: 0 auto;
  max-width: ${px2rem(980)};
  padding: 0 ${px2rem(40)} ${px2rem(80)} ${px2rem(40)};

  h3 {
    margin-bottom: ${px2rem(4)};
  }
`;
