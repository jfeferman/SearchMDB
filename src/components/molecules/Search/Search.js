import React from 'react';
import {
  Cta,
  TextInput,
} from '../../atoms';
import { StyledSearch } from './styles';
import { CONTENT_KEYS } from '../../../config/constants';

const Search = () => {
  return (
    <StyledSearch>
      <div><TextInput placeholder={CONTENT_KEYS.SEARCH_PLACEHOLDER} /></div>
      <div><Cta>Search</Cta></div>
    </StyledSearch>
  );
};

export default Search;
