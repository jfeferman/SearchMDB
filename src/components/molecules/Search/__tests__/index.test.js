import React from 'react';
import 'jest-styled-components';
import { create } from 'react-test-renderer';
import Search from '..';
import {
  Cta,
  TextInput,
} from '../../../atoms';
import { StyledSearch } from '../styles';

describe('components/molecules/Search', () => {
  let renderer;
  let instance;

  beforeEach(() => {
    renderer = create(<Search />);
    instance = renderer.root;
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('renders a container', () => {
    expect(instance.findAll(el => el.type === StyledSearch)).toHaveLength(1);
  });

  it('renders one text input', () => {
    expect(instance.findAll(el => el.type === TextInput)).toHaveLength(1);
  });

  it('renders one cta', () => {
    expect(instance.findAll(el => el.type === Cta)).toHaveLength(1);
  });
});
