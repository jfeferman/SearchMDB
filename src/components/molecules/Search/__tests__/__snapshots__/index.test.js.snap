// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Search matches the snapshot of the component 1`] = `
.c2 {
  border-radius: 0.625rem;
  border: 1px solid;
  cursor: pointer;
  display: block;
  font-weight: 300;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  width: auto;
  background: #DDDDDD;
  color: #000000;
  font-size: 0.9375rem;
  line-height: 1.0625rem;
  -webkit-letter-spacing: 0.02125rem;
  -moz-letter-spacing: 0.02125rem;
  -ms-letter-spacing: 0.02125rem;
  letter-spacing: 0.02125rem;
  padding: 1rem 2.59375rem;
  vertical-align: middle;
  outline: 0;
  -webkit-transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out;
}

.c2:focus {
  box-shadow: 0px 0px 2px 2px rgba(23,109,220,0.8);
}

.c1 {
  width: 100%;
  box-sizing: border-box;
  height: 3.125rem;
  border: 1px solid #DDDDDD;
  border-radius: 0.1875rem;
  background-color: #ffffff;
  padding-left: 1.25rem;
  font-size: 1rem;
  min-width: 25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0 auto;
  width: 37.5rem;
}

.c0 div:nth-of-type(1) {
  margin-right: 0.625rem;
}

<div
  className="c0"
>
  <div>
    <input
      className="c1"
      placeholder="Search for movie, tv show, person..."
      type="text"
    />
  </div>
  <div>
    <button
      className="c2"
      disabled={false}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
